src/logging.jl

```diff
@@ -24,20 +42,123 @@ function MPILogger(iostream, ctx::AbstractCommsContext)
 end
 
 """
-    MPIFileLogger(context, log_dir)
+    FileLogger(context, log_dir; log_stdout = true, min_level = Logging.Info)
 
 Log MPI ranks to different files within the `log_dir`.
+
+The minimum logging level is set using `min_level`.
+If `log_stdout = true`, root process logs will be sent to stdout as well.
 """
-function MPIFileLogger(
-    ctx::AbstractCommsContext,
+function FileLogger(
```

Comment:
Can we change this to

```julia
function FileLogger(
    io::IO,
    ctx::SingletonCommsContext,
    log_dir::AbstractString;
    log_stdout = true,
    min_level::Logging.LogLevel = Logging.Info,
)

FileLogger(
    ctx::SingletonCommsContext,
    log_dir::AbstractString;
    log_stdout = true,
    min_level::Logging.LogLevel = Logging.Info,
) = FileLogger(stdout, ctx, log_dir; log_stdout, min_level)
```
so that we can support `FileLogger(stdout, ctx, log_dir; log_stdout, min_level) do io end`.